### PR TITLE
Add proxy authentication support, fixes #102

### LINKED ISSUE #102: Authentication for HTTP proxies

### PR DESCRIPTION
**Pull request by kayhuba**
Initial implementation to support forward proxies with authentication.

Some notes:
* The AHC 1.9 branch is having an issue regarding proxy authentication: https://github.com/AsyncHttpClient/async-http-client/issues/996 - In order to get it working I had to upgrade to the AHC 2.0 branch
* AHC 2.0 requires JDK8. Thus I increased the sourceCompatibility level to 1.8

I am aware that this may be a difficult requirement, yet it was required to make it work. The workaround in https://github.com/AsyncHttpClient/async-http-client/issues/996#issuecomment-172879211 did not work out eventually - at least not with the test setup I had locally (Apache forward proxy with basic authentication).

Looking forward to your feedback.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,9 +3,7 @@ language: java
 matrix:
   include:
   - dist: precise
-    jdk: openjdk7
-  - dist: precise
-    jdk: oraclejdk7
+    jdk: oraclejdk8
```
Comment (aiham) [inline .travis.yml]: This is unnecessary since we already have oraclejdk8 in the test matrix (See line 8)

**Comment by kayhuba** [comment]
Thanks for your feedback @aiham. I've updated the PR according to your notes. Somehow there's a test failing now, which I can't really relate to my changes. Is this maybe a glitch with the test? Is it possible to trigger another Travis build and see if it works then?

**Comment by aiham** [comment]
@kayhuba Actually could you get rid of `- dist: precise`? We only needed precise for Java 7.

Also now that I've merged your other PR, you'll need to update your `getArchives(String sessionId)` method, particularly removing the reference to `IOException`.

Thanks

**Comment by kayhuba** [comment]
@aiham Both done (` - dist: precise` removed, `IOException` removed). Trevor CI checks have passed.

**Comment by aiham** [comment]
Thanks @kayhuba. I'll publish 4.0.0 next week.

**Comment by aiham** [comment]
@kayhuba I've published v4.0.0

**Comment by kayhuba** [comment]
Thanks a lot @aiham !

**Pull request merged**